src/client.cpp

```diff
@@ -939,6 +948,37 @@ void CClient::AudioCallback ( CVector<int16_t>& psData, void* arg )
     // get the pointer to the object
     CClient* pMyClientObj = static_cast<CClient*> ( arg );
 
+    QMutexLocker locker(&pMyClientObj->MutexAudioProcess);
+    if ( pMyClientObj->Sound.IsRunning() == false )
+    {
+        return;
+    }
+
+    // In common audio subsystems, there may be multiple audio sizes in use.
```

Comment:
Interesting concept. Is this Linux-specific or does this apply to all supported OSs?

Reply:
It applies to all OS'es where Jamulus is not the only client of the sound card subsystem.

Usually sound mixers have a pre-defined and fixed interval for the buffering. 

Also further, USB hardware only supports small buffers of size of 2ms 3ms 4ms and so on. When Jamulus specifies 64 samples of buffer, the USB hardware cannot technically program that! So you end up with audio bursts, that in turn affect the timing of the UDP packets, and I found this to be the root cause of all jitter problems during my tests with FreeBSD. Too bad Windows doesn't have a nanosecond clock. I'm not fully sure how this patch will work there. Maybe someone can help testing?

Reply:
I can do it when I am back at home in a week. If Windows does not support it, we should move your code in the Jack Sound.cpp file for now. Actually with my audio devices I tested, I did not have any higher jitter from the audio driver. What audio hardware are you using? And how did you measure the jitter?

Reply:
I have a X32-RACK audio device configured with 8-channel audio, where I use 2 for Jamulus.  I measured the jitter by using wireshark and looking at the time between data packets going on the wire.